### PR TITLE
Updated coredns version and added README

### PR DESCRIPTION
**Pull request by kschumy**
*Issue #, if available:*

*Description of changes:*
* Updated coredns version to be 1.8.4 for all versions.
* Added README about updating coredns
* Updated attribution
* Updated CHECKSUMS
* Updated Golang version: https://github.com/coredns/coredns/blob/053c4d5ca1772517746a854e87ffa971249df14b/go.mod#L3
* EDIT: removed if condition related to kops in cluster_wait.sh  and deleted core_dns_cluster_role.yaml

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jaxesn — CHANGES_REQUESTED** [review]
This looks pretty good. Not sure why the checksums arent matching, curious how you generated these?

Youll also need to change this, https://github.com/aws/eks-distro/blob/main/development/kops/cluster_wait.sh#L41, probably can just remove the if all together now that its the same for all versions.

**Comment by kschumy** [comment]
> This looks pretty good. Not sure why the checksums arent matching, curious how you generated these?
> 
> Youll also need to change this, https://github.com/aws/eks-distro/blob/main/development/kops/cluster_wait.sh#L41, probably can just remove the if all together now that its the same for all versions.

We're going to apply the patch added to 1.20 and 1.21 to 1.18 and 1.19. See  PR #547. So for this PR, I removed the if condition related to kops in cluster_wait.sh  and deleted core_dns_cluster_role.yaml

**Comment by kschumy** [comment]
/ hold

The versions may not be correct. I'm looking into this

**Comment by kschumy** [comment]
/hold

See last comment

File: projects/coredns/coredns/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 BASE_DIRECTORY=$(shell git rev-parse --show-toplevel)
 GIT_TAG?=$(shell cat $(RELEASE_BRANCH)/GIT_TAG)
-GOLANG_VERSION?=1.13
+GOLANG_VERSION?=1.16
```
Comment (jaxesn) [inline projects/coredns/coredns/Makefile:3]:
```suggestion
GOLANG_VERSION?=$(shell cat $(RELEASE_BRANCH)/GOLANG_VERSION)
```

File: projects/coredns/coredns/Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 BASE_DIRECTORY=$(shell git rev-parse --show-toplevel)
 GIT_TAG?=$(shell cat $(RELEASE_BRANCH)/GIT_TAG)
-GOLANG_VERSION?=1.13
+GOLANG_VERSION?=$(shell cat ./$(RELEASE_BRANCH)/GOLANG_VERSION)
```
Comment (jaxesn) [inline projects/coredns/coredns/Makefile:3]: Remove the ./ Just to be consistent with the one above.

File: projects/coredns/coredns/1-19/ATTRIBUTION.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 
 ** cloud.google.com/go/compute/metadata; version v0.56.0 --
-https://cloud.google.com/go
+https://github.com/googleapis/google-cloud-go
```
Comment (jaxesn) [inline projects/coredns/coredns/1-19/ATTRIBUTION.txt:3]: Rever this change

**Review by jaxesn — CHANGES_REQUESTED** [review]

**Review by jaxesn — APPROVED** [review]
/lgtm

**Pull request merged**